### PR TITLE
chore(deps): upgrade to symfony 4.4

### PR DESCRIPTION
**Pull request by victorvassilev**
## Description

Update package version of symfony components from 4.3 to 4.4

**Fixes** #MON-4968
## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

You can do a new composer install and see the product working fine.

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: src/Centreon/Application/Validation/CentreonValidatorFactory.php [hunk 1]
```diff
@@ -73,11 +74,16 @@ public function getInstance(Constraint $constraint)
 
         if (!isset($this->validators[$className])) {
             if (class_exists($className)) {
-                // validator as a class with dependencies
-                $this->validators[$className] = new $className(new ServiceLocator(
-                    $this->container,
-                    method_exists($className, 'dependencies') ? $className::dependencies() : []
-                ));
+                // validator as a class with dependencies from centreon
+                if (in_array(CentreonValidatorInterface::class, class_implements($className))) {
```
Comment (vhr) [inline src/Centreon/Application/Validation/CentreonValidatorFactory.php]: please use the same approach as here: https://github.com/centreon/centreon/blob/master/src/CentreonRemote/Infrastructure/Service/ExporterService.php#L18

File: src/Centreon/Application/Validation/Validator/Interfaces/CentreonValidatorInterface.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * do not wish to do so, delete this exception statement from your version.
+ *
+ * For more information : contact@centreon.com
+ *
+ *
+ */
+
+namespace Centreon\Application\Validation\Validator\Interfaces;
+
+interface CentreonValidatorInterface
```
Comment (vhr) [inline src/Centreon/Application/Validation/Validator/Interfaces/CentreonValidatorInterface.php]: will be helpful if the interface has docblock with information about its purpose

Comment (victorvassilev) [reply src/Centreon/Application/Validation/Validator/Interfaces/CentreonValidatorInterface.php]: thanks adding info now

File: src/Centreon/Application/Validation/Validator/Interfaces/CentreonValidatorInterface.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+{
+
```
Comment (kduret) [inline src/Centreon/Application/Validation/Validator/Interfaces/CentreonValidatorInterface.php]: Could you add `validate` method to this interface

File: src/Centreon/Application/Validation/CentreonValidatorFactory.php (hunk 1 above)
Comment (victorvassilev) [reply src/Centreon/Application/Validation/CentreonValidatorFactory.php]: thanks

File: src/Centreon/Application/Validation/Validator/Interfaces/CentreonValidatorInterface.php (hunk 3 above)
Comment (victorvassilev) [reply src/Centreon/Application/Validation/Validator/Interfaces/CentreonValidatorInterface.php]: This method is required by this interface, which the classes implement: centreon/centreon-bam/vendor/symfony/validator/ConstraintValidatorInterface.php. The purpose of this interface is to determine if the validator is a centreon custom one that has dependencies or is a generic one from symfony/validator

File: composer.json [hunk 4]
```diff
@@ -101,12 +101,14 @@
...
-            "require": "4.3.*"
+            "require": "4.4.*"
         }
     },
     "scripts": {
         "auto-scripts": {
-            "security-checker security:check": "script"
+            "security-checker security:check": "script",
+            "cache:clear": "symfony-cmd",
+            "assets:install %PUBLIC_DIR%": "symfony-cmd"
```
Comment (callapa) [inline composer.json]: What is the purpose of this command?

Comment (victorvassilev) [reply composer.json]: it got auto-added with the composer install, looks like auto-checker for dependencies:
https://github.com/sensiolabs/security-checker

**Pull request merged**